### PR TITLE
Removing autovalue from audio feature

### PR DESCRIPTION
**Pull request by tomaszpolanski**

### CODE REVIEW

File: app/src/main/java/com/futurice/freesound/feature/audio/DefaultObservableExoPlayer.java [hunk 1]
```diff
@@ -89,6 +89,7 @@
     }
 
     private static boolean isTimelineChanging(@NonNull final ExoPlayerState playerState) {
-        return playerState.playbackState() == ExoPlayer.STATE_READY && playerState.playWhenReady();
+        return playerState.getPlaybackState() == ExoPlayer.STATE_READY && playerState
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/audio/DefaultObservableExoPlayer.java]: This should wrap differently, before the `&&`.

File: app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerState.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2016 Futurice GmbH
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerState.kt]: Check your copyright template. This should be using 2017 now.

File: app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java [hunk 3]
```diff
@@ -56,8 +56,8 @@ protected void subscribeActual(final Observer<? super ExoPlayerState> observer)
         observer.onSubscribe(listener);
         exoPlayer.addListener(listener);
         if (emitInitial) {
-            emitValue(ExoPlayerState.create(exoPlayer.getPlayWhenReady(),
-                                            exoPlayer.getPlaybackState()),
+            emitValue(new ExoPlayerState(exoPlayer.getPlayWhenReady(),
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java]: Formatting

Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java]: I get a warning about `this.exoPlayer` being `@NonNull` but not initialized by `ExoPlayerStateObservable`'s package visible constructor. Do you get the same?

File: app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java [hunk 4]
```diff
@@ -39,6 +39,7 @@
 import polanski.option.Option;
 
 import static com.futurice.freesound.common.utils.Preconditions.get;
+import static com.futurice.freesound.feature.audio.IdKt.from;
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java]: This `Kt` suffix is a great incentive to use Kotlin everywhere!

File: app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java [hunk 5]
```diff
@@ -125,8 +126,8 @@ void openDetails() {
     }
 
     void toggleSoundPlayback() {
-        audioPlayer.togglePlayback(PlaybackSource.create(Id.from(sound.id()),
-                                                         sound.previews().lowQualityMp3Url()));
+        audioPlayer.togglePlayback(new PlaybackSource(from(sound.id()),
+                                                                   sound.previews().lowQualityMp3Url()));
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java]: Formatting

File: app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java [hunk 6]
```diff
@@ -135,8 +136,9 @@ void toggleSoundPlayback() {
     }
 
     private boolean isThisSound(@NonNull final PlayerState playerState) {
-        return playerState.source()
-                          .filter(playbackSource -> playbackSource.id().equals(Id.from(sound.id())))
+        return playerState.getSource()
+                          .filter(playbackSource -> playbackSource.getId().equals(
+                                  from(sound.id())))
```
Comment (peter-tackage) [inline app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java]: `from(sound.id())))` could be on previous line

File: app/src/test/java/com/futurice/freesound/feature/audio/DefaultObservableExoPlayerTest.java [hunk 7]
```diff
@@ -164,7 +164,8 @@ ArrangeBuilder withEndedExoPlayer() {
         }
 
         ArrangeBuilder withExoPlayerStateStreamEvent(boolean playWhenReady, int state) {
-            exoPlayerStateOnceAndStream.onNext(ExoPlayerState.create(playWhenReady, state));
+            exoPlayerStateOnceAndStream.onNext(
+                   new ExoPlayerState(playWhenReady, state));
```
Comment (peter-tackage) [inline app/src/test/java/com/futurice/freesound/feature/audio/DefaultObservableExoPlayerTest.java]: Unnecessarily wrapped.

Comment (peter-tackage) [inline app/src/test/java/com/futurice/freesound/feature/audio/DefaultObservableExoPlayerTest.java]: I get an error/warning here about "use of Kotlin internal declaration from different module".

File: app/src/test/java/com/futurice/freesound/feature/audio/ExoPlayerAudioPlayerTest.java [hunk 8]
```diff
@@ -314,7 +314,7 @@ ArrangeBuilder withEndedExoPlayer() {
         }
 
         ArrangeBuilder withExoPlayerStateStreamEvent(boolean playWhenReady, int state) {
-            exoPlayerStateStream.onNext(ExoPlayerState.create(playWhenReady, state));
+            exoPlayerStateStream.onNext(new ExoPlayerState(playWhenReady, state));
```
Comment (peter-tackage) [inline app/src/test/java/com/futurice/freesound/feature/audio/ExoPlayerAudioPlayerTest.java]: Same "internal from different module" warning here

File: app/src/test/java/com/futurice/freesound/feature/search/SoundItemViewModelTest.java [hunk 9]
```diff
@@ -358,10 +359,10 @@ public void progressPercentage_isNone_whenDifferentSoundActiveInPlayer_usingId()
                                 .url(url1)
                                 .build();
         new ArrangeBuilder()
-                .withPlayerStateEvent(PlayerState.create(PlayerState.State.PLAYING,
-                                                         Option.ofObj(PlaybackSource
-                                                                              .create(Id.from(id2),
-                                                                                      url1))));
+                .withPlayerStateEvent(new PlayerState(State.PLAYING,
+                                                      Option.ofObj(
```
Comment (peter-tackage) [inline app/src/test/java/com/futurice/freesound/feature/search/SoundItemViewModelTest.java]: Odd formatting here and other places.  Could be:

```java
        new ArrangeBuilder()
                .withPlayerStateEvent(new PlayerState(State.PLAYING,
                                                      Option.ofObj(new PlaybackSource(from(id2),
                                                                                      url1))));
```

File: app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerState.kt (hunk 2 above)
Comment (tomaszpolanski) [inline app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerState.kt]: Kotlin converter created the file

File: app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java (hunk 3 above)
Comment (tomaszpolanski) [inline app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java]: Now I don't. I am using Canary 5 version, maybe that changes something

File: app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java (hunk 4 above)
Comment (tomaszpolanski) [inline app/src/main/java/com/futurice/freesound/feature/search/SoundItemViewModel.java]: I comes out of the box

**Pull request merged**